corehq/apps/case_search/management/commands/stale_es_case_search_check.py

```diff
@@ -0,0 +1,288 @@
+                try:
+                    for case_search in fetch_case_searches(list(cases), query_size):
+                        case_searches += 1
+                        case_id = case_search["_id"]
+                        try:
+                            smo_cs = string_to_datetime(case_search["server_modified_on"])
+                        except KeyError:
+                            smo_cs = None
+                        smo_case = cases.pop(case_id)
+                        try:
+                            delta = smo_case - smo_cs
+                        except TypeError:
+                            try:
+                                delta = smo_case.replace(tzinfo=smo_cs.tzinfo) - smo_cs
+                            except TypeError:
+                                delta = None
+                        # we only care about case_searches whose
+                        # `server_modified_on` is *older* (greater than)
+                        if delta is None or delta > warn_case_newer_than:
+                            table.add_row([
+                                domain,
+                                case_id,
+                                human_td(delta),
+                                human_dt(smo_case),
+                                human_dt(smo_cs),
+                            ])
+                            mismatches += 1
+                    if cases:
+                        logger.warning("found %s cases missing from case_search index", len(cases))
+                        for case_id, smo_case in cases.items():
+                            table.add_row([
+                                domain,
+                                case_id,
+                                "n/a",
+                                human_dt(smo_case),
+                                "n/a",
+                            ])
+                finally:
+                    logger.info("fetched %s case_searches", case_searches)
+                    logger.info("found %s mismatched records for domain: %s", mismatches, domain)
+            logger.info("done.")
+        finally:
+            if options["csv"]:
+                table.write_csv(options["output"])
+            else:
+                options["output"].write(table.render())
+
+
+def domain_subset(domains, key):
+    if len(key) not in (1, 2):
+        raise ValueError(f"invalid length (must be 1 or 2 hex digits): {key}")
+    key = (key * 2) if len(key) == 1 else key
+    min = int(key[0], 16)
+    max = int(key[1], 16)
+    subset = []
+    for name in domains:
+        index = int(md5(name.encode("utf-8")).hexdigest()[0], 16)
+        if min <= index and index <= max:
+            subset.append(name)
+    return subset
+
+
+def get_case_ids_and_mod_escases(domain):
+    query = (CaseES()
+        .domain(domain)
+        .source(["_id", "case_id", "server_modified_on"])
+        .size(CASE_SCROLL_SIZE))
+    for case in query.scroll():
```

Comment:
I thought I saw somewhere that `.scroll()` sometimes returned docs without `_id` for some unknown reason. Is that a potential issue here?